bot/exts/backend/error_handler.py

```diff
@@ -0,0 +1,157 @@
+            self.revert_cooldown_counter(ctx.command, ctx.message)
+            embed = self.error_embed(
+                f"Your input was invalid: {error}\n\nUsage:\n"
+                f"```{ctx.prefix}{ctx.command} {ctx.command.signature}```"
+            )
+            await ctx.send(embed=embed)
+            return
+
+        elif isinstance(error, commands.CommandOnCooldown):
+            mins, secs = divmod(math.ceil(error.retry_after), 60)
+            embed = self.error_embed(
+                f"This command is on cooldown:\nPlease retry in **{mins} minutes {secs} seconds**.",
+                NEGATIVE_REPLIES,
+            )
+            await ctx.send(embed=embed, delete_after=10)
+            return
+
+        elif isinstance(error, commands.DisabledCommand):
+            await ctx.send(
+                embed=self.error_embed(
+                    "This command has been disabled.", NEGATIVE_REPLIES
+                )
+            )
+            return
+
+        elif isinstance(error, commands.NoPrivateMessage):
+            await ctx.send(
+                embed=self.error_embed(
+                    "This command can only be used in the server.", NEGATIVE_REPLIES
+                )
+            )
+            return
+
+        elif isinstance(error, commands.BadArgument):
+            self.revert_cooldown_counter(ctx.command, ctx.message)
+            embed = self.error_embed(
+                "The argument you provided was invalid: "
+                f"{error}\n\nUsage:\n```{ctx.prefix}{ctx.command} {ctx.command.signature}```"
+            )
+            await ctx.send(embed=embed)
+            return
+        else:
+            await self.handle_unexpected_error(ctx, error)
+            return  # Exit early to avoid logging.
+
+    async def handle_unexpected_error(
+        self, ctx: commands.Context, error: commands.CommandError
+    ) -> None:
+        """Send a generic error message in `ctx` and log the exception as an error with exc_info."""
+        await ctx.send(
+            f"Sorry, an unexpected error occurred. Please let us know!\n\n"
+            f"```{error.__class__.__name__}: {error}```"
+        )
+
+        push_alert = Embed(
+            title="An unexpected error occurred",
+            color=Colours.soft_red,
+        )
+        push_alert.add_field(
+            name="User",
+            value=f"id: {ctx.author.id} | username: {str(ctx.author)}",
+            inline=False,
+        )
+        push_alert.add_field(
+            name="Command", value=ctx.command.qualified_name, inline=False
+        )
+        push_alert.add_field(
+            name="Message & Channel",
+            value=f"Message: {ctx.message.id} | Channel {ctx.channel.id}",
+            inline=False,
+        )
+        push_alert.add_field(
+            name="Full Message", value=ctx.message.content, inline=False
+        )
+
+        dev_alerts = self.bot.get_channel(Channels.devalerts)
+        if not dev_alerts:
```

Comment:
```suggestion
        if dev_alerts is None:
```
You could be more explicit here, too.

Reply:
Done!